src/hooks/subscriptions/useMessagesLane.ts

```diff
@@ -41,33 +41,31 @@ const useMessagesLane = ({ isApiReady, api, chain }: SubscriptionInput): Output
   const laneId = useLaneId();
   const { bridgedMessages } = getSubstrateDynamicNames(chain);
   const isReady = !!(isApiReady && api.query[bridgedMessages] && chain);
-
   const getOutboundLaneData = useCallback(
     () =>
-      api.query[bridgedMessages].outboundLanes(laneId, (res: any) => {
-        const latest_generated_nonce = res.get('latest_generated_nonce').toString();
-        const latest_received_nonce = res.get('latest_received_nonce').toString();
-        const pendingMessages = new BN(latest_generated_nonce).sub(new BN(latest_received_nonce));
-
-        setOutboudLanes({
-          latestReceivedNonce: latest_received_nonce.toString(),
-          pendingMessages: pendingMessages.isNeg() ? '0' : pendingMessages.toString(),
-          totalMessages: latest_generated_nonce
-        });
+      getLaneData({
```

Comment:
Enclosing this code into functions would make sense if we avoided touch `api` directly, but few lines above we still do `api.query[bridgedMessages]` for instance.

I think ideally we wouldn't even avoid calling into `getSubstrateDynamicNames` here. Basically the entire:
1. polkadot.js/api logic
2. any runtime-specific calls
would be abstracted away for these kind of components.

Of course it's easier said than done, so what I'm saying here is that this is not incorrect, but I'd like to challenge you to go a bit beyond this.

Reply:
Supporting what tomek said , in PR #186 there is this file that is already taking into account what he propose: https://github.com/paritytech/parity-bridges-ui/pull/186/files#diff-7de2939806355f65b85bb93c1220b85ffe6a7f55fb0eb9aef1a7f7295aabdd8b

Reply:
> but few lines above we still do api.query[bridgedMessages] for instance.

As far as I understand the `api.query[bridgedMessages` is actually not necessary at that line. I will push an update now that fixes it and even maybe accomplishes the challenge - but there is a glitch that I do not like so much